### PR TITLE
AV-805: Better harvester reporting

### PR DESCRIPTION
**Pull request by bzar**
- Better information from `harvester_status`
- Filtering options for `harvester_status`, default to skipping manual and never run harvesters
- Better error handling for `opendata-harvest send-status-emails`
- Command line options for `opendata-harvest send-status-emails`

### CODE REVIEW

File: modules/ckanext-ytp_main/ckanext/ytp/harvesterstatusplugin.py [hunk 1]
```diff
@@ -17,21 +17,39 @@ def last_job_status(source):
...
+
+        if finished is not None:
+            status = 'finished'
+        elif started is not None:
+            status = 'running'
         else:
-            status = 'error'
+            status = 'pending'
+
+        status = 'running' if finished is None else 'finished'
```
Comment (Aaronkala) [inline modules/ckanext-ytp_main/ckanext/ytp/harvesterstatusplugin.py]: This is a leftover?

Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/harvesterstatusplugin.py]: Yes, fixed.

File: modules/ckanext-ytp_main/ckanext/ytp/commands.py [hunk 2]
```diff
@@ -482,52 +482,76 @@ def add_to_groups(ctx, config, dryrun):
...
+            'finished': '%%(title)-%ds | Finished %%(time)s with %%(errors)d errors',
+            'pending': '%%(title)-%ds | Pending since %%(time)s'}
+    unknown_status_template = '%%(title)-%ds | Unknown status: %%(status)s'
+    max_title_length = max(len(title) for title in status)
+
+    def status_string(title, values):
+        template = status_templates.get(values.get('status'), unknown_status_template)
+        status = values.get('status')
+        time_field = 'finished' if status == 'finished' else 'started'
+        return template % max_title_length % {
```
Comment (Aaronkala) [inline modules/ckanext-ytp_main/ckanext/ytp/commands.py]: What does the max_title_length here do?

Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/commands.py]: It adjusts the title column width to be the same for all status rows so they're aligned.

Comment (Aaronkala) [reply modules/ckanext-ytp_main/ckanext/ytp/commands.py]: Ok nice didn't know that syntax :)

**Pull request merged**